### ISSUE TITLE
Alternative mdns feature using system's avahi/bonjour

### ISSUE THREAD

**Issue by eskimor** [libp2p/rust-libp2p#1815]
I am working on this, creating this ticket to track it.

In addition to the current mdns implementation I am working on an alternative implementation taking advantage of an already running bonjour/avahi implementation. For systems where those services are already present, this comes with a couple of advantages:

- More standard conform implementation which is battle tested and solid
- More feature completeness (IPv6 for example, support for multiple interfaces, ...)
- Basically maintenance free (Maintenance is done in the daemons itself)
- Better system performance as only one daemon on the system is handling mdns packets.

The current implementation should stay as a fallback especially for systems not yet supported by the [zeroconf](https://crates.io/crates/zeroconf) crate.

**Comment by thomaseizinger** [comment]
Let me know if you want to reopen, closing as stale.
